### PR TITLE
Stepper Feature

### PR DESCRIPTION
**Pull request by sibteali786**
## Describe your changes
- Trying to fix activeStepIndex to support expressions 
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

**Comment by sibteali786** [comment]
@evshi need some help here, first when I update step using Next it does not work, and remains on old value ? Second is it a right approach what i am doing here to make a separate hook for this ?

**Comment by sibteali786** [comment]
The example is at `/widgets` screen at bottom

**Review by evshi — COMMENTED** [review]
I think what you're looking for is just a `useEffect` hook that syncs the evaluated attribute to the component local state

File: packages/runtime/src/widgets/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -51,35 +54,39 @@ interface CustomConnectorProps {
 }
 
 const Stepper: React.FC<StepperProps> = (props) => {
-  const [activeStep, setActiveStep] = useState(props.activeStepIndex ?? 0);
+  // Explicit update to evaluate activeStepIndex
+  const resVals = useEvaluate({
+    activeStepIndex: props.activeStepIndex,
+  });
```
Comment (evshi) [inline packages/runtime/src/widgets/Stepper/Stepper.tsx:60]: Remove this. Just move the useEffect hook after useRegisterBindings.

File: packages/runtime/src/widgets/Stepper/Stepper.tsx [hunk 2]
```diff
@@ -32,7 +35,7 @@ export interface StepProps {
 
 export type StepperProps = {
   steps: StepProps[];
-  activeStepIndex?: number;
```
Comment (evshi) [inline packages/runtime/src/widgets/Stepper/Stepper.tsx:35]: This can still be null?

File: packages/runtime/src/widgets/Stepper/Stepper.tsx [hunk 3]
```diff
@@ -51,35 +53,35 @@ interface CustomConnectorProps {
...
-      handleNext,
-      handleBack,
-    },
-  );
+  const { values, rootRef } = useRegisterBindings({ ...props }, props.id, {
+    handleNext,
+    handleBack,
+  });
+  useEffect(() => {
+    if (isString(activeStep) && isExpression(activeStep)) {
```
Comment (evshi) [inline packages/runtime/src/widgets/Stepper/Stepper.tsx:80]: You can check the props value so you don't need to overload your state variable. Your state variable should always just be an integer

**Review by evshi — APPROVED** [review]

File: packages/runtime/src/widgets/Stepper/Stepper.tsx [hunk 4]
```diff
@@ -16,14 +16,16 @@ import {
   useRegisterBindings,
   useTemplateData,
   unwrapWidget,
+  isExpression,
 } from "@ensembleui/react-framework";
 import type { EnsembleWidget, Expression } from "@ensembleui/react-framework";
 import Alert from "@mui/material/Alert";
+import { isString } from "antd/es/button";
```
Comment (evshi) [inline packages/runtime/src/widgets/Stepper/Stepper.tsx:23]: wrong import use from lodash-es

**Review by evshi — CHANGES_REQUESTED** [review]

**Comment by sibteali786** [comment]
there is one more issue happening here @evshi i forgot to discuss it yesterday, the state of stepper is not being updated when we change a step, it does change step but not the value of activeStepIndex nor stiveStep . This way it always stays what it was initialized with ?

**Pull request merged**